src/components/Form/Form.tsx

```diff
@@ -0,0 +1,151 @@
+import "./style.scss";
+import { useFormik } from "formik";
+import { Modal } from "../../interfaces";
+import { films } from "../../films";
+import { CONSTANTS } from "../../constants";
+import Select from "react-select";
+import DeleteForm from "./DeleteForm";
+
+const selectStyle: any = {
+  menu: () => {
+    return {
+      backgroundColor: "#232323;",
+      position: "absolute",
+      margin: "20px",
+      marginTop: "0",
+      display: "block",
+      width: "330px",
+      top: "48px",
+      borderRadius: "3px",
+    };
+  },
+  option: () => {
+    return {
+      ":hover": {
+        backgroundColor: "#f65261;",
+      },
+    };
+  },
+  control: () => {
+    return {
+      backgroundColor: "#555",
+      margin: "20px",
+      border: "none",
+      height: "26px",
+      borderRadius: "3px",
+      color: "white",
+      display: "flex",
+    };
+  },
+};
+
+interface FormProps {
+  closeModal(): void;
+  modalState: Modal;
+}
+
+const Form: React.FC<FormProps> = ({ closeModal, modalState }) => {
+  const movie: any =
+    modalState.type === CONSTANTS.FORM_TYPE.ADD
+      ? CONSTANTS.ADD_MOVIE
+      : films.find((el) => {
+          return el.id === modalState.filmID;
+        });
+
+  const formik = useFormik({
+    initialValues: movie,
+    onSubmit: (values) => {
+      values.releaseDate = new Date(values.releaseDate);
+      if (modalState.type === CONSTANTS.FORM_TYPE.EDIT) {
+        films[films.indexOf(movie)] = values;
+      } else if (modalState.type === CONSTANTS.FORM_TYPE.DELETE) {
+        films.splice(films.indexOf(movie), films.indexOf(movie));
+      } else {
+        films.push(values);
+      }
+      closeModal();
+    },
+    onReset: (values) => {
+      values = movie;
+    },
+  });
+  return (
+    <div className="form-wrapper">
+      <form
+        className="form"
+        onSubmit={formik.handleSubmit}
+        onReset={formik.handleReset}
+      >
+        <div className="close" onClick={closeModal}></div>
+        <h3>{modalState.type.toUpperCase()} MOVIE</h3>
+        {modalState.type === CONSTANTS.FORM_TYPE.DELETE ? (
+          <DeleteForm />
+        ) : (
+          <>
+            <label htmlFor="title">TITLE</label>
+            <input
+              id="title"
+              name="title"
```

Comment:
a lot duplication, create constants for these all inputs fileds title, releaseDate,... etc.